A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+has both subchannels and channels, the channels are not delegated to by the 
+superchannel.  An example would be a channel that is used by the ancestor 
+channel, but does not handle RPCs given by the application.
+
+Each channel has a ChannelRef reference which includes the channel id and an
+optional name.  The name is included for human consumption.  There are no
+restrictions on the name but it should be limited to a reasonable length.
+
+Each subchannel has a SubchannelRef reference which includes the subchannel id
+and an optional name.  The name is included for human consumption.  There are 
+no restrictions on the name but it should be limited to a reasonable length.
+
+![channel hierarchy 1][channel hierarchy 1]
+
+### Channel Data
+
+Channels and Subchannels include data about themselves, notably their initial
+parameters and their call stats.  Currently the data includes:
+
+* Channel state ( See: connectivity-semantics-and-api.md )
+* Channel trace
+* Channel target, if applicable.
+* Number of calls starts, succeeded, and failed.  Note this is slightly
+  different than the number of streams.  See the Socket Data section.
+* Time of the last call started.
+
+In general, each piece of data included is specific to the channel itself and
+NOT of its descendent channels.  This is to say that an ancestor channel
+is not an aggregator for descendent channels.  The target of a channel or
+subchannel may be different from that of its parent, or not present at all.  The
+channel state may be different from the ancestor channel state.   The number of
+calls started, succeeded, and failed should reflect if calls were specifically
+tied to the channel.
+
+The number of calls started, succeeded, and failed on a channel gives insight
+into the activity of the channel.  Subtracting the number of failed and
+succeeded calls from the number started indicates how many are in flight.  The
+total number of started calls indicates how active the channel has been over
+time.  The ratio between succeeded calls and failed calls gives a rough
+estimation of how healthy the channel has been.  The last call started timestamp
+indicates how close the channel is towards entering the idle state.
+
+#### Channel Trace
+
+Channels and Subchannels also have a concept of a channel trace.  This includes
+interesting events that have happened on the channel recently.  Example events
+include entering different channel states, name resolution, load balancing
+changes, unreachability, etc.  As mentioned earlier, this tracing info is
+specific to the channel, and not an aggregation of descendent channel traces.
+
+The channel traces may include ids pointing to channels and subchannels outside
+of the current DAG.  Since channel traces use the same ids as the channels 
+themselves, this allows a user to query for more information about a channel 
+given its trace.  For example, the trace may mention that name resolution 
```

Comment:
I don't think this is a good example.  In general, I think traces for name resolution will only be present in the channel that does the name resolution.

I think the only cases where a trace would contain an entry that points to a node other than the current node is when a parent channel references an owned channel (like the grpclb channel) or a subchannel.  And in those cases, it would generally point to another node in the same tree.  The only edge case I can think of would be an entry that points to a subchannel that has subsequently been removed from the tree (e.g., because that address has been removed by a resolver update).

Reply:
Not so clear.  The channel that triggers name resolution may want to indicate that here.  

This is really only a stub section, and will need to be clarified when the Channel Trace design is written.   This section is informative, but not normative.

Reply:
I cannot think of a case where we would actually want to do this.  I think this example is misleading. 
 Please either remove this example or change it to something more realistic.